clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -161,6 +160,7 @@ public RecordAppendResult append(TopicPartition tp,
                                      long timestamp,
                                      byte[] key,
                                      byte[] value,
+                                     long expectedOffset,
```

Comment:
I'm not sure if you chose `-1L` because the type here is `long` instead of `Long` but the difference between this and the other values is that, to me, this is an optional setting so I'd recommend to use `Long` everywhere